srtp/srtp.c

```diff
@@ -2604,6 +2604,8 @@ srtp_err_status_t srtp_unprotect_mki(srtp_ctx_t *ctx,
         /* now compute auth function over packet */
         status = srtp_auth_update(session_keys->rtp_auth, (uint8_t *)auth_start,
                                   *pkt_octet_len - tag_len - mki_size);
+        if (status)
+            return status;
```

Comment:
If `srtp_auth_update()` above fails, existing code returns the `status`. But if `srtp_auth_compute()` below fails, existing code returns `srtp_err_status_auth_fail` rather than the original status code. So I was not sure whether to best return `status` or `srtp_err_status_auth_fail` here and arbitrarily chose the former. This applies to all changes in this file. Please tell me if you disagree.